<?xml version="1.0"?>
<launch>
    <include file="$(find-pkg-share panda_controller)/launch/controller.launch.xml"/>

    <node pkg="panda_controller" exec="slider_controller.py"/>

    <node pkg="joint_state_publisher_gui" exec="joint_state_publisher_gui"
                    name="joint_state_publisher" output="screen">
        <remap from="/joint_states" to="joint_commands"/>
    </node>
    <!-- remapping은 joint_states라는 토픽의 이름을 단순히 joint_conmmands라는 이름으로 바꿀 뿐임 -->
    <!-- ros2control은 현재 joint의 위치를 joint_states 토픽으로 발행하기 때문에, gui에서 joint_states라는 이름으로
    topic을 발행하는 경우 겹침. 따라서 remapping을 통해 토픽이름을 바꿈. -->
    <!-- slider_controller.py에서 구독하는 토픽 이름도 joint_commands로 하기 -->
</launch>
